tools/c7n_gcp/c7n_gcp/resources/logging.py

```diff
@@ -18,20 +18,133 @@
 # how to map them given a project level root entity sans use of c7n-org
 
 
-@resources.register('logsink')
-class LogSink(QueryResourceManager):
+@resources.register('log-sink')
+class LoggingSink(QueryResourceManager):
 
     class resource_type(TypeInfo):
         service = 'logging'
-        version = 'v1'
+        version = 'v2'
         component = 'projects.sinks'
         enum_spec = ('list', 'sinks[]', None)
         scope_key = 'parent'
-        scope_template = "projects/{}/sinks"
+        scope_template = "projects/{}"
         id = "name"
 
         @staticmethod
         def get(client, resource_info):
-            return client.get('get', {
+            return client.execute_query('get', {
                 'sinkName': 'projects/{project_id}/sinks/{name}'.format(
                     **resource_info)})
+
+
+@resources.register('log-project-sink')
+class LoggingProjectSink(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.sinks'
+        enum_spec = ('list', 'sinks[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {
+                'sinkName': 'projects/{project_id}/sinks/{name}'.format(
+                    **resource_info)})
+
+
+@resources.register('log-project-metric')
+class LoggingProjectMetric(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.metrics'
+        enum_spec = ('list', 'metrics[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'name'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query('get', {
+                'metricName': 'projects/{project_id}/metrics/{name}'.format(
+                    **resource_info)})
+
+
+@resources.register('log-project-log')
+class LoggingProject(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'logging'
+        version = 'v2'
+        component = 'projects.logs'
+        enum_spec = ('list', 'logNames[]', None)
+        scope_key = 'parent'
+        scope_template = 'projects/{}'
+        id = 'id'
+
+    def augment(self, resources):
+        out_resources = []
+        for resource in resources:
+            type, id, key, log_id = resource.split('/')
+            out_resources.append({
+                'type': type,
+                'id': id,
```

Comment:
this is not the id of the log objects, its a project id..

Reply:
Updated